pkg/apis/ceph.rook.io/v1/object.go

```diff
@@ -45,8 +45,20 @@ func (s *ObjectStoreSpec) GetPort() (int32, error) {
 	return -1, errors.New("At least one of Port or SecurePort should be non-zero")
 }
 
+// IsExternal returns true if at least one `v1.EndpointAddress` is specified, and all specified `v1.EndpointAddress`
+// in the `ExternalRgwEndpoints` point to a non-empty IP address.
 func (s *ObjectStoreSpec) IsExternal() bool {
-	return len(s.Gateway.ExternalRgwEndpoints) != 0
+	isExternal := len(s.Gateway.ExternalRgwEndpoints) != 0
```

Comment:
Even if this struct is list, AFAIR RGW supports only one IP address at least the current code works like that. (@leseb please confirm), so if len is more than > 1, IMO it is error cases

Reply:
What makes you think we only support a single IP address?

Reply:
@leseb: Above assumption which I made because I saw one k8s service, one k8s endpoint generated, one port can be used(both http/https values can be given), so that it can one represent only one external RGW server.

Reply:
Assuming all the RGW are listening on the same port, we can have a single service and N endpoints. The endpoint is a list IIRC.